### PR TITLE
Schedule deadman notification when ENF is disabled (EXPOSUREAPP-13154)

### PR DESCRIPTION
**Pull request by mtwalli**
## Changes 
- Schedule Deadman notification despite ENF being disabled
- Fallback to App install time to calculate delay in calculation in case ENF is disabled on App onboarding and calculation info is available 
## To help testing this easily 
1- Change calculation minutes to `5` or any number you choose 
```diff
- const val DEADMAN_NOTIFICATION_DELAY_MINUTES = 36 * DateTimeConstants.MINUTES_PER_HOUR
+ const val DEADMAN_NOTIFICATION_DELAY_MINUTES = 5 //36 * DateTimeConstants.MINUTES_PER_HOUR
```
2- Schedule `PeriodicWorkRequestBuilder` to run every 15 min
- There should be one notification since app start and ENF is disabled 

[Ticket](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-13154)

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/deadman/DeadmanNotificationScheduler.kt [hunk 1]
```diff
@@ -21,30 +19,17 @@ class DeadmanNotificationScheduler @Inject constructor(
...
-            )
-            isOnboarded && isTracingEnabled
-        }.onEach { shouldSchedulePeriodic ->
-            Timber.d("shouldSchedulePeriodic: $shouldSchedulePeriodic")
-            if (shouldSchedulePeriodic) {
-                schedulePeriodic()
-            } else {
-                cancelScheduledWork()
-            }
-        }.launchIn(appScope)
```
[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/deadman/DeadmanNotificationScheduler.kt:47]: You are right, relying on onboarding flag only won't cancel the workers, but as I understood from the logic the one time worker will get replaced all the time with the latest delay of calculation. I think we should still cancel the workers somehow.

[the remark this answers is not in the document]
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/deadman/DeadmanNotificationScheduler.kt:47]: @chiljamgossow  I updated it now, please have at PR description ?

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt [hunk 2]
```diff
@@ -20,6 +20,7 @@ import dagger.Provides
 import de.rki.coronawarnapp.CoronaWarnApplication
 import de.rki.coronawarnapp.util.SafeNavDeepLinkBuilder
 import de.rki.coronawarnapp.util.worker.WorkManagerProvider
+import org.joda.time.Instant
```
Comment (chiljamgossow) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt:23]: java.time?

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/util/di/AndroidModule.kt]: Updated

**Review by chiljamgossow — APPROVED** [review]

**Pull request merged**